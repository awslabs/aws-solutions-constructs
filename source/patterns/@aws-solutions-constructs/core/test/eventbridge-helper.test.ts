/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import { Stack } from 'aws-cdk-lib';
import * as events from 'aws-cdk-lib/aws-events';
import * as defaults from '../index';
import { Template } from 'aws-cdk-lib/assertions';

// --------------------------------------------------------------
// Test deployment with no properties
// --------------------------------------------------------------
test('Test deployment with no properties', () => {
  // Stack
  const stack = new Stack();
  // Helper declaration
  defaults.buildEventBus(stack, {
    eventBusProps: {
      eventBusName: 'testneweventbus'
    }
  });

  const template = Template.fromStack(stack);
  template.resourceCountIs("AWS::EventBridge::EventBus", 0);
});

// --------------------------------------------------------------
// Test deployment with existing EventBus
// --------------------------------------------------------------
test('Test deployment with existing EventBus', () => {
  // Stack
  const stack = new Stack();
  // Helper declaration
  defaults.buildEventBus(stack, {
    existingEventBusInterface: new events.EventBus(stack, `existing-event-bus`, { eventBusName: 'test-bus' })
  });

  Template.fromStack(stack).resourceCountIs('AWS::Events::EventBus', 1);
});

// --------------------------------------------------------------
// Test deployment with new EventBus with props
// --------------------------------------------------------------
test('Test deployment with new EventBus with props', () => {
  // Stack
  const stack = new Stack();
  // Helper declaration
  defaults.buildEventBus(stack, {
    eventBusProps: {
      eventBusName: 'testneweventbus'
    }
  });

  Template.fromStack(stack).hasResourceProperties('AWS::Events::EventBus', {
    Name: 'testneweventbus'
  });
});