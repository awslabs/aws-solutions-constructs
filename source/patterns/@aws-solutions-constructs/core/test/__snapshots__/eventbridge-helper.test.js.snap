// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test deployment with existing EventBus 1`] = `
Object {
  "Resources": Object {
    "existingeventbusA5B80487": Object {
      "Properties": Object {
        "Name": "existingeventbus",
      },
      "Type": "AWS::Events::EventBus",
    },
  },
}
`;

exports[`Test deployment with new EventBus no props 1`] = `
Object {
  "Resources": Object {
    "CustomEventBusEC0C3CB8": Object {
      "Properties": Object {
        "Name": "CustomEventBus",
      },
      "Type": "AWS::Events::EventBus",
    },
  },
}
`;

exports[`Test deployment with new EventBus with props 1`] = `
Object {
  "Resources": Object {
    "testneweventbus87AFE59D": Object {
      "Properties": Object {
        "Name": "testneweventbus",
      },
      "Type": "AWS::Events::EventBus",
    },
  },
}
`;

exports[`Test deployment with no properties 1`] = `Object {}`;
